server/models.py

```diff
@@ -0,0 +1,581 @@
+
+
+class BaseModel(Base):
+    __abstract__ = True
+    created_at = Column(
+        UTCDateTime, default=lambda: dt.now(timezone.utc), nullable=False
+    )
+    updated_at = Column(
+        UTCDateTime,
+        default=lambda: dt.now(timezone.utc),
+        onupdate=lambda: dt.now(timezone.utc),
+        nullable=False,
+    )
+
+
+# on-delete-cascade is done in SQLAlchemy like this:
+# https://stackoverflow.com/questions/5033547/sqlalchemy-cascade-delete
+
+
+class Organization(BaseModel):
+    id = Column(String(200), primary_key=True)
+    name = Column(String(200), nullable=False, unique=True)
+
+    elections = relationship(
+        "Election",
+        back_populates="organization",
+        passive_deletes=True,
+        order_by="Election.election_name",
+    )
+
+
+# Election is a slight misnomer - this model represents an Election.
+class Election(BaseModel):
+    id = Column(String(200), primary_key=True)
+
+    election_name = Column(String(200), nullable=False)
+    election_date = Column(String(40), nullable=False)
+    polls_open = Column(String(5), nullable=False)
+    polls_close = Column(String(5), nullable=False)
+    polls_timezone = Column(String(4), nullable=False)
+    certification_date = Column(String(40), nullable=False)
+
+    # Who does this election belong to?
+    organization_id = Column(
+        String(200),
+        ForeignKey("organization.id", ondelete="cascade"),
+        nullable=False
+    )
+    organization = relationship("Organization", back_populates="elections")
+
+    jurisdictions = relationship(
+        "Jurisdiction",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="Jurisdiction.name",
+    )
+
+    # The jurisdictions file contains a list of jurisdictions
+    # and emails for the admins of each jurisdiction.
+    # We use this to create Jurisdictions and JAs.
+    jurisdictions_file_id = Column(
+        String(200), ForeignKey("file.id", ondelete="set null")
+    )
+    jurisdictions_file = relationship(
+        "File",
+        foreign_keys=[jurisdictions_file_id],
+        single_parent=True,
+        cascade="all, delete-orphan",
+    )
+
+    # The definition file contains election related data
+    definition_file_id = Column(
+        String(200), ForeignKey("file.id", ondelete="set null")
+    )
+    definition_file = relationship(
+        "File",
+        foreign_keys=[definition_file_id],
+        single_parent=True,
+        cascade="all, delete-orphan",
+    )
+
+    # Contests in the election
+    contests = relationship(
+        "Contest",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="Contest.name",
+    )
+
+    # Parties in the election
+    parties = relationship(
+        "Party",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="Party.name",
+    )
```

Comment:
is this needed?

Reply:
@benadida this can be removed if needed, it's just there since it was in the election definition file.